src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/SetAzureStorageAccount.cs

```diff
@@ -120,9 +132,53 @@ public bool EnableHttpsTrafficOnly
                 enableHttpsTrafficOnly = value;
             }
         }
-
         private bool? enableHttpsTrafficOnly = null;
 
+        [Parameter(HelpMessage = "Whether to set Storage Account Encryption KeySource to Microsoft.Storage or not.", Mandatory = false, ParameterSetName = StorageEncryptionParameterSet)]
+        public SwitchParameter StorageEncryption
+        {
+            get { return storageEncryption; }
+            set { storageEncryption = value; }
+        }
+        private bool storageEncryption = false;
+
+        [Parameter(HelpMessage = "Whether to set Storage Account encryption keySource to Microsoft.Keyvault or not. " +
+            "If you specify KeyName, KeyVersion and KeyvaultUri, Storage Account Encryption KeySource will also be set to Microsoft.Keyvault weather this parameter is set or not.", 
+            Mandatory = false, ParameterSetName = KeyvaultEncryptionParameterSet)]
+        public SwitchParameter KeyvaultEncryption
+        {
+            get { return keyvaultEncryption; }
+            set { keyvaultEncryption = value; }
+        }
+        private bool keyvaultEncryption = false;
+
+        [Parameter(HelpMessage = "Storage Account encryption keySource KeyVault KeyName",
+                    Mandatory = true,
+                    ParameterSetName = KeyvaultEncryptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyName { get; set; }
+
+        [Parameter(HelpMessage = "Storage Account encryption keySource KeyVault KeyVersion",
+        Mandatory = true,
+        ParameterSetName = KeyvaultEncryptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyVersion { get; set; }
+
+        [Parameter(HelpMessage = "Storage Account encryption keySource KeyVault KeyVaultUri",
+        Mandatory = true,
+        ParameterSetName = KeyvaultEncryptionParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string KeyVaultUri
+        {
+            get; set;
+        }
+
+        [Parameter(
+        Mandatory = false,
+        HelpMessage = "Storage Account Identity Type.")]
+        [ValidateSet("SystemAssigned",IgnoreCase = true)]
+        public string IdentityType { get; set; }
```

Comment:
If user don't specify the para, the account won't have ID. The ID is only used in Enablde Keyvault encryption scenarios only now, and used for Keyvault to give permission to the Account.. We don't set the Idenetity Type by default since we don't want to enable it for account which don't need it.
We required a value since we think there might be other value added to the parameter, so require a value to avoid breaking change. The value SystemAssigned means the ID is from system generated